owasp-top10-2017-apps/a2/saidajaula-monster/app/app.py

```diff
@@ -102,15 +117,38 @@ def login():
         if not password.validate_password(result[0]):
             return "Login failed! \n"
 
-        cookie_dic = {"permissao": result[1], "username": form_username}
-        cookie = json.dumps(cookie_dic)
-        hash_cookie = hashlib.sha256(cookie.encode('utf-8')).hexdigest()
-        cookie_done = '.'.join([cookie,hash_cookie])
-        cookie_done = base64.b64encode(str(cookie_done).encode("utf-8"))
+        access_token, refresh_token = create_tokens(form_username, result[1])
+
         resp = make_response("Logged in!")
-        resp.set_cookie("sessionId", cookie_done)
+        resp.set_cookie("access_token", access_token)
+        resp.set_cookie("refresh_token", refresh_token, path="/refresh-token")
         return resp
 
+@app.route("/refresh-token", methods=['GET'])
```

Comment:
I would name this route as `/access-token`, because what we should do here is to generate a new access token, based on the current refresh token. The HTTP request done by the front-end would become something like
```
GET /access-token
```
which sounds like: `Hey, server, give me a new access token, please!`